src/GraphQL/Transports/IGraphQLTextSerializer.cs

```diff
@@ -0,0 +1,26 @@
+using GraphQL.Transports.Json;
+
+namespace GraphQL
+{
+    /// <summary>
+    /// Serializes and deserializes object hierarchies to/from a string.
+    /// Should include special support for <see cref="ExecutionResult"/>, <see cref="Inputs"/>
+    /// and transport-specific classes as necessary.
+    /// Typical JSON-specific classes are providied within <see cref="Transports.Json">GraphQL.Transports.Json</see>.
+    /// </summary>
+    public interface IGraphQLTextSerializer : IGraphQLSerializer
+    {
+        /// <summary>
+        /// Serializes the specified object to a string and returns it.
+        /// Typically used to write <see cref="ExecutionResult"/> instances to a JSON result.
+        /// </summary>
+        string Serialize<T>(T? value);
+
+        /// <summary>
+        /// Deserializes the specified string to the specified object type.
+        /// Typically used to parse <see cref="GraphQLRequest"/> instances from JSON.
+        /// A <paramref name="value"/> of <see langword="null"/> returns <see langword="default"/>.
+        /// </summary>
+        T Deserialize<T>(string? value);
```

Comment:
I had these as `Read`/`Write` but that just doesn't make sense when converting to/from a string.

Reply:
Write for string is string allocation (create).

Reply:
I fail to understand your meaning; please explain.

What I was trying to say that the term `Write` in `string Write<T>(T? value)` is improper since it is not writing to a string, as one would a StringBuilder, but rather serializing internally and returning the string as the result, which `Serialize` conveys better.

Reply:
I tried to say the same.